mobile/components/views/common/components/details/review/ReviewsModal.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Alert, FlatList, Modal, Text, TouchableOpacity, View } from 'react-native';
+import { reviewModal } from './styles';
+import {
+  fetchAttraction,
+  fetchAttractionOpinions,
+} from '../../../../Attractions/api/attractionsApi';
+import { Attraction, Opinion } from '../../../../Attractions/types';
+import { attractionsExamples, reviewsExamples } from '../../../../Attractions/api/fake/apiMock';
+import ReviewTile from './ReviewTile';
+import AttractionReviewTile from './AtrractionReviewTile';
+import SortTile from './SortTile';
+import FilterTile from './FilterTile';
+
+interface ReviewsModalProps {
+  attractionId: number;
+  visible: boolean;
+  hideMenu: () => void;
+  onNewReview: () => void;
+  onEditReview: (review: Opinion) => void;
+}
+
+const ReviewsModal: React.FC<ReviewsModalProps> = ({
+  attractionId,
+  visible,
+  hideMenu: onHide,
+  onNewReview,
+  onEditReview,
+}) => {
+  const [attraction, setAttraction] = useState<Attraction>();
+  const [attractionOpinions, setAttractionOpinions] = useState<Opinion[]>([]);
+  const flatListRef = useRef<FlatList<Opinion>>(null);
```

Comment:
It's important to note that useRef provides access to the actual DOM elements rather than React's virtual DOM. This means that changes directly made to these elements may not be visible to React, potentially causing the component not to re-render in response to these changes. While useRef is a powerful tool for persisting values across renders without triggering a re-render itself, its interaction with the actual DOM elements needs careful consideration. Direct manipulations might not always reflect in React's virtual DOM, leading to discrepancies. Although it's not certain that issues will arise in every case, it's crucial to keep this behavior in mind when working with useRef and the actual DOM.

Reply:
Okay, I'll remember that. Here it seems to work fine.